### PR TITLE
Support for socksproxytype plugins in Settings dialog

### PR DESCRIPTION
**Pull request by g1itch**
Hello.

With this changes now it's possible to set "Proxy server / Tor Type:" to any one provided by plugin. `start_proxyconfig` will be called when user press "OK" in the Settings dialog.

This is a draft because I'm trying to fix test and maybe port in `proxyconfig_stem`.

### CODE REVIEW

**Comment by dimyme** [comment]
> it's possible to set "Proxy server / Tor Type:" to any one provided by plugin:

meaning i.e.:  proxy_type in SOCKS4a, SOCKS5

any one of these two, choose freely!

**Comment by g1itch** [comment]
> > it's possible to set "Proxy server / Tor Type:" to any one provided by plugin:
> 
> meaning i.e.: proxy_type in SOCKS4a, SOCKS5
> 
> any one of these two, choose freely!

Seriously? tl;dr!

Actually, I meant 'stem' or any other plugin for the `bitmessage.proxyconfig` entry point. Another silly comment and I'll report abuse.

File: src/helper_startup.py [hunk 1]
```diff
@@ -30,14 +33,14 @@ def loadConfig():
         needToCreateKeysFile = config.safeGet(
             'bitmessagesettings', 'settingsversion') is None
         if not needToCreateKeysFile:
-            print(
+            logger.info(
```
Comment (PeterSurda) [inline src/helper_startup.py]: Good, these are the few places where I left print due to initialisation order, which you now fixed.

**Comment by g1itch** [comment]
I made this "Ready for review" because it contains the solution for periodic Travis CI fails, e.g. https://travis-ci.org/Bitmessage/PyBitmessage/builds/629056869: I separated clearnet and tor tests and marked tor tests for skipping (because stem package isn't installed).

Manual testing scenario I used for this changeset (and it fails):
 1. on first start select "Let me configure network..." in Connection dialog;
 1. set Proxy server / Tor Type to 'stem'
 1. pybitmessage should connect to the network through tor and start accepting incoming connections
 1. go back to the network settings and set Proxy server / Tor Type to 'none'
 1. pybitmessage should connect to the network directly using the same port 8444

**Pull request merged**